subsys/net/lib/shell/iface.c

```diff
@@ -817,6 +818,47 @@ static int cmd_net_default_iface(const struct shell *sh, size_t argc, char *argv
 	return 0;
 }
 
+static int cmd_net_link_speed(const struct shell *sh, size_t argc, char *argv[])
+{
+#if !defined(CONFIG_NET_L2_ETHERNET_MGMT)
+       PR_WARNING("Unsupported command, please enable CONFIG_NET_L2_ETHERNET_MGMT\n");
+       return -ENOEXEC;
+#else
+       struct ethernet_req_params params = { 0 };
+       int idx = get_iface_idx(sh, argv[1]);
+       uint16_t speed = atoi(argv[2]);
+       struct net_if *iface;
+
+       if (idx < 0) {
+               return -ENOEXEC;
+       }
+
+       iface = net_if_get_by_index(idx);
+       if (!iface) {
+               PR_WARNING("No such interface in index %d\n", idx);
+               return -ENODEV;
+       }
+
```

Comment:
We need to make sure that the interface is Ethernet one. So something like this (see similar checks in other parts of this file).
```
	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
               PR_WARNING("Interface %idx is not Ethernet type\n", idx);
               return -EINVAL;
       }
```